tests/gui/helper.py

```diff
@@ -191,40 +190,40 @@ def _get_children(self, w):
 		elif hasattr(w, 'findChildren'):
 			return w.findChildren()
 
-	def _find(self, name):
+	def _find(self, widgets, name):
 		"""Recursive find a widget by name.
 
 		This is the actual search implementation behind `GuiHelper.find`.
+
+		Finds all components that match the right-most name, e.g. foo in bar/baz/foo. From
+		there, go up the tree, removing candidates that have no match for other path
+		components, e.g. no baz in their parents.
 		"""
-		match = None
-		seen = set()
-		widgets = deque(self.active_widgets)
+		path_components = name.split('/')
 
-		path_components = list(reversed(name.split('/')))
+		first_part = path_components.pop()
+		filtered = [w for w in widgets if w.name == first_part]
+		if not filtered:
+			return None
 
+		candidates = [(f, [f.parent]) for f in filtered]
```

Comment:
Looking at pychan's code, lots of things seem to point towards a single parent.

`pychan.widgets.widget.Widget`

```python
		# Parent attribute makes sure we only have one parent,
		# that tests self.__parent - so make sure we have the attr here.
		self.__parent = None
		self.parent = parent

```

When you add a child to a widget (say a Container), you use `addChild` which does

```python
		widget.parent = self
```

But yes, it seems possible that a widget ends up in two `children` lists, however it will only ever have one parent. If it ends up in multiple lists I'd consider it a bug on our side.